staging/src/k8s.io/mount-utils/mount_linux_test.go

```diff
@@ -715,19 +699,110 @@ func TestFormat(t *testing.T) {
 			}
 			mounter := NewSafeFormatAndMount(nil, exec, WithMaxConcurrentFormat(tc.max, tc.timeout))
 
-			var wg sync.WaitGroup
+			for i := 0; i < tc.max+1; i++ {
+				go func() {
+					mounter.format(fstype, nil)
+				}()
+			}
+
+			time.Sleep(100 * time.Millisecond)
+
+			mu.Lock()
+			if concurrent != tc.max {
+				t.Errorf("SafeFormatAndMount.format() got concurrency: %d, want: %d", concurrent, tc.max)
+			}
+			mu.Unlock()
+
+			for i := 0; i < tc.max; i++ {
```

Comment:
comment
// signal the commands to finish the goroutines, this will allow the command
// that is pending to be executed

Reply:
resolved in 0a21ba0476